tsig.go

```diff
@@ -39,6 +39,14 @@ type TSIG struct {
 	OtherData  string `dns:"size-hex:OtherLen"`
 }
 
+// TsigProvider provides the API to plug-in a custom TSIG implementation.
+type TsigProvider interface {
+	// Generate is passed the DNS message to be signed and partial TSIG RR and returns the MAC bytes and any error.
+	Generate([]byte, *TSIG) ([]byte, error)
+	// Verify is passed the DNS message to be verified and TSIG RR and returns any error.
```

Comment:
This probably needs to be expanded slightly. Something explaining it should only return `nil` if the signature is valid and an error otherwise.

Reply:
Done.